File: projects/grind-75/typescript/array/two-sum.ts
export function twoSum(nums: number[], target: number): number[] {
  const store = new Map<number, number>();
  for (let [i, num] of nums.entries()) {
    if (store.has(num)) {
      return [store.get(num)!, i];
    }
    store.set(target - num, i);
  }
  return [];
}

File: projects/grind-75/go/array/two_sum/two_sum.go
package util

func twoSum(nums []int, target int) []int {
	store := make(map[int]int)
	for i, num := range nums {
		if val, ok := store[num]; ok {
			return []int{val, i}
		}
		store[target-num] = i
	}
	return nil
}
